carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import pandas as pd
+from sklearn import preprocessing
+
+from ..api import MLModel
+from .load_model import load_model
+
+
+class MLModelCatalog(MLModel):
+    def __init__(
+        self, data, data_name, model_type, ext="h5", cache=True, models_home=None, **kws
+    ):
+        """
+        Constructing the ML model
+
+        Parameters
+        ----------
+        model_type : str
+            Name of the model ``{name}.{ext}`` on https://github.com/indyfree/cf-models.
+        data_name : str
+            Name of the dataset the model has been trained on.
+        cache : boolean, optional
+            If True, try to load from the local cache first, and save to the cache
+            if a download is required.
+        models_home : string, optional
+            The directory in which to cache data; see :func:`get_models_home`.
+        kws : keys and values, optional
+            Additional keyword arguments are passed to passed through to the read model function
+        data : data.api.Data Class
+            Correct dataset for ML model
+        ext : String
+            File extension of saved ML model file
+        """
+        # check if dataset fits to ML model
+        assert data.name == data_name
```

Comment:
Asserts are mostly used for testing or to give the source code reader for program flow. Here is a good stackoverflow link of usecases.

tl;dr: throw execption when you want to check for cases that could potentially happen (like in this case)

Reply:
On another note, in this specific case we could just pass the DataCatalog object and use the data.name object from there instead of _also_ passing data_name in the contructor

Reply:
The assertion is fixed in PR #17 and should be disappear after merging.
You're right, data_name is unnecessary and I replaced it with data.name in the next commit.